src/genegraph/database/dataset.clj

```diff
@@ -0,0 +1,98 @@
+(ns genegraph.database.dataset
+  "Namespace for handling operations on persistent Jena datasets.
+  Specifically designed around handling asychronous writes. "
+  (:require [clojure.spec.alpha :as spec]
+            [clojure.core.async :as async])
+  (:import [org.apache.jena.tdb2 TDB2Factory]
+           [org.apache.jena.query Dataset ReadWrite]
+           [java.util.concurrent BlockingQueue ArrayBlockingQueue TimeUnit]
+           [java.util List ArrayList]
+           [org.apache.jena.query.text TextDatasetFactory]))
+
+
+(defrecord PersistentDataset [dataset
```

Comment:
I know defrecord doesn't take a doc string, but it would be nice to describe the args here

Reply:
Yup, per above comment would like to use spec for this.